File: benchmarks/async/02_fanout.go
package main

import (
	"fmt"
	"sync"
)

func compute(n int) int {
	total := 0
	for i := 1; i <= n; i++ {
		total += i
	}
	return total
}

func main() {
	results := make([]int, 100)
	var wg sync.WaitGroup
	for i := 0; i < 100; i++ {
		wg.Add(1)
		go func(idx int) {
			defer wg.Done()
			results[idx] = compute(idx + 1)
		}(i)
	}
	wg.Wait()
	total := 0
	for _, v := range results {
		total += v
	}
	fmt.Printf("checksum = %d\n", total)
}

File: benchmarks/async/02_fanout.py
"""Async benchmark 2: Fan-out (Python asyncio baseline)."""

import asyncio


async def compute(n: int) -> int:
    total = 0
    for i in range(1, n + 1):
        total += i
    return total


async def fanout() -> int:
    tasks = [compute(i) for i in range(1, 101)]
    results = await asyncio.gather(*tasks)
    return sum(results)


def main():
    result = asyncio.run(fanout())
    print(f"checksum = {result}")


if __name__ == "__main__":
    main()
